src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +36,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
     (ActorFunc)BgSpot18Basket_Update,
     (ActorFunc)BgSpot18Basket_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7710.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7770.s")
+static ColliderJntSphItemInit sJntSphItemsInit[2] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+        { 0, { { 0, 2040, 0 }, 54 }, 100 },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { 1, { { 0, 1400, 0 }, 13 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x09, 0x20, COLSHAPE_JNTSPH },
+    2,
+    sJntSphItemsInit,
+};
+
+static s16 D_808B85C8[] = { 0x8000, 0x2AAA, 0xD555, 0x0000 };
+
+void func_808B7710(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+
+    Collider_InitJntSph(globalCtx, &this->colliderJntSph);
+    Collider_SetJntSph(globalCtx, &this->colliderJntSph, &this->dyna.actor, &sJntSphInit, &this->colliderJntSphItems);
+    this->dyna.actor.colChkInfo.mass = 0xFF;
+}
+
+void func_808B7770(BgSpot18Basket* this, GlobalContext* globalCtx, f32 arg2) {
+    Vec3f acceleration;
+    Vec3f velocity;
+    Vec3f position;
+    f32 cosValue;
+    s32 i = 0;
+    f32 randomValue;
+    static s16 D_808B85D0 = 0;
+    f32 sinValue;
+    s32 numberOfLoops = 2;
+
+    while (i != numberOfLoops) {
```

Comment:
In general FryJay, your names often seem to describe what the code does on an "interpreter" level (numberOfLoops, sinValue) rather than reflecting the "intended" usage (respectively effectCount, dirX for example), which isn't very useful, you can just keep names like temp_v0 and sp34 if you have no idea like it often happens to everyone lol